variation/regex.py

```diff
@@ -0,0 +1,215 @@
+)
+
+# (#_#)_(#_#) OR (?_#)_(#_?)
+GENOMIC_DELETION_AMBIGUOUS_1 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+# (?_#)_#, (#_?)_# OR (#_#)_#
+GENOMIC_DELETION_AMBIGUOUS_2 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_(?P<pos2>\d+)del$"
+)
+
+# #_(#_?) OR #_(#_#)
+GENOMIC_DELETION_AMBIGUOUS_3 = re.compile(
+    r"^(?P<pos0>\d+)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+# _REGEXPRS are used to help group the regex pattern and associated token type and
+# classification type
+
+# Note: Order matters for regexprs
+PROTEIN_REGEXPRS: List[Tuple[any, TokenType, ClassificationType]] = [
+    (PROTEIN_DELINS, TokenType.PROTEIN_DELINS, ClassificationType.PROTEIN_DELINS),
+    (PROTEIN_DELETION, TokenType.PROTEIN_DELETION, ClassificationType.PROTEIN_DELETION),
+    (
+        PROTEIN_SUBSTITUTION,
+        TokenType.PROTEIN_SUBSTITUTION,
+        ClassificationType.PROTEIN_SUBSTITUTION,
+    ),
+    (
+        PROTEIN_REFERENCE_AGREE,
+        TokenType.PROTEIN_REFERENCE_AGREE,
+        ClassificationType.PROTEIN_REFERENCE_AGREE,
+    ),
+    (
+        PROTEIN_INSERTION,
+        TokenType.PROTEIN_INSERTION,
+        ClassificationType.PROTEIN_INSERTION,
+    ),
+]
+
+# Note: Order matters for regexprs
+CDNA_REGEXPRS: List[Tuple[any, TokenType, ClassificationType]] = [
+    (CDNA_GENOMIC_DELINS, TokenType.CDNA_DELINS, ClassificationType.CDNA_DELINS),
+    (CNDA_GENOMIC_DELETION, TokenType.CDNA_DELETION, ClassificationType.CDNA_DELETION),
+    (
+        CDNA_GENOMIC_SUBSTITUTION,
+        TokenType.CDNA_SUBSTITUTION,
+        ClassificationType.CDNA_SUBSTITUTION,
+    ),
+    (
+        CDNA_GENOMIC_REFERENCE_AGREE,
+        TokenType.CDNA_REFERENCE_AGREE,
+        ClassificationType.CDNA_REFERENCE_AGREE,
+    ),
+    (
+        CDNA_GENOMIC_INSERTION,
+        TokenType.CDNA_INSERTION,
+        ClassificationType.CDNA_INSERTION,
+    ),
+]
+
+# Note: Order matters for regexprs
+GENOMIC_REGEXPRS: List[Tuple[any, TokenType, ClassificationType]] = [
+    (CDNA_GENOMIC_DELINS, TokenType.GENOMIC_DELINS, ClassificationType.GENOMIC_DELINS),
+    (
+        CNDA_GENOMIC_DELETION,
+        TokenType.GENOMIC_DELETION,
+        ClassificationType.GENOMIC_DELETION,
+    ),
+    (
+        CDNA_GENOMIC_SUBSTITUTION,
+        TokenType.GENOMIC_SUBSTITUTION,
+        ClassificationType.GENOMIC_SUBSTITUTION,
+    ),
+    (
+        CDNA_GENOMIC_REFERENCE_AGREE,
+        TokenType.GENOMIC_REFERENCE_AGREE,
+        ClassificationType.GENOMIC_REFERENCE_AGREE,
+    ),
+    (
+        CDNA_GENOMIC_INSERTION,
+        TokenType.GENOMIC_INSERTION,
+        ClassificationType.GENOMIC_INSERTION,
+    ),
+    (
+        GENOMIC_DUPLICATION,
+        TokenType.GENOMIC_DUPLICATION,
+        ClassificationType.GENOMIC_DUPLICATION,
+    ),
+]
+
+
+# Note: Order matters for regexprs
+GENOMIC_DUP_AMBIGUOUS_REGEXPRS: List[
+    Tuple[any, TokenType, ClassificationType, AmbiguousType]
```

Comment:
are the annotations on here and 195 supposed to be `AmbiguousRegexType`?

Reply:
Yes